src/main/java/com/online/store/controllers/product/controllers/AddProductController.java

```diff
@@ -0,0 +1,32 @@
+package com.online.store.controllers.product.controllers;
+
+import com.online.store.lib.Injector;
+import com.online.store.model.Product;
+import com.online.store.service.ProductService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AddProductController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("com.online.store");
+    private final ProductService productService =
+            (ProductService) injector.getInstance(ProductService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/products/add.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws IOException {
+        String name = req.getParameter("name");
+        double price = Double.parseDouble(req.getParameter("price"));
```

Comment:
Let's create a check for the input data
We can't work with the empty Strings

Reply:
this should be done by jsp 
https://mate-academy.github.io/jv-program-fulltime/02_jdbc_and_servlets/checklist/hw_16_1.html

Reply:
done

Reply:
done